pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -99,13 +99,13 @@ func (p *provider) getConfig(s runtime.RawExtension) (*Config, *providerconfig.C
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	c.Username, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.Username)
+	c.Username, err = p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.Username, "OS_USERNAME")
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, fmt.Errorf("failed to get the value of \"username\" field, error = %v", err)
 	}
-	c.Password, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.Password)
+	c.Password, err = p.configVarResolver.GetConfigVarStringValueOrEnv(rawConfig.Password, "OS_PASSWORD")
 	if err != nil {
-		return nil, nil, nil, err
+		return nil, nil, nil, fmt.Errorf("failed to get the value of \"password\" field, error = %v", err)
 	}
 	c.DomainName, err = p.configVarResolver.GetConfigVarStringValue(rawConfig.DomainName)
```

Comment:
Could you also add `OS_AUTH_URL` for `identityEndpoint`, `OS_TENANT_NAME` for `tenantName` and `OS_DOMAIN_NAME` for `domainName` so we have the full set of auth-related env vars?

Reply:
Many thanks for pointing this out. I missed them.